### ISSUE TITLE
Problem with electron launch  from within atom

### ISSUE THREAD

**Issue by Minyar2004** [platformio/platformio-atom-ide-terminal#290]
Hi,

It is impossible to launch my Electron application using your package "platformio-atom-ide-terminal" within atom, I have this error message:

```
PS D:\electron\MyProject> npm start

> MyProject@1.0.0 start D:\electron\MyProject
> electron .

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! MyProject@1.0.0 start: `electron .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the DLWEB-Browser@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2017-07-25T15_41_03_909Z-debug.log
```

However, when I launch the same project on Windows PowerShell, it works properly.

I tried to launch the Quick Start example given in https://electron.atom.io/docs/tutorial/quick-start/

Same problem, impossible to run it from the atom terminal

can you help me fix this error?

**Comment by andersdigital** [comment]
I can confirm I am having a similar issue with electron. Doing "electron ." results in the terminal re-initialising. I can confirm this happens with platformio-ide-terminal set to PowerShell and Git Bash.
PowerShell and Git Bash will launch electron fine as stand alone ~~applications~~ terminals.

**Comment by webskydavid** [comment]
I can confirm the same problem. On Git Bash, Windows CMD and PowerShell as a separete program work great. Overrideing the shell to C:\Windows\System32\cmd.exe or GIT Bash gives the same error.

**Comment by andersdigital** [comment]
I have attached the contents of npm-debug.log below. I have hashed out any information pertaining to my project

    0 info it worked if it ends with ok
    1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
    1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
    1 verbose cli   'start' ]
    2 info using npm@3.10.10
    3 info using node@v6.10.3
    4 verbose run-script [ 'prestart', 'start', 'poststart' ]
    5 info lifecycle ########@0.0.1~prestart: ########@0.0.1
    6 silly lifecycle ########@0.0.1~prestart: no script for prestart, continuing
    7 info lifecycle ########@0.0.1~start: ########@0.0.1
    8 verbose lifecycle ########@0.0.1~start: unsafe-perm in lifecycle true
    9 verbose lifecycle ########@0.0.1~start: PATH: C:\Program Files\nodejs\node_modules\npm\bin\node-\ngyp-bin;D:\########\node_modules\.bin;C:\Program Files (x86)\Git\mingw32\bin;C:\Program Files (x86)\Git\usr\bin;########l\bin;C:\Program Files (x86)\OpenSSH\bin;C:\Program Files\nodejs;C:\Program \nFiles (x86)\Skype\Phone;C:\Users\########\AppData\Local\Microsoft\WindowsApps;C:\Program Files (x86)\Nmap;C:\Program Files (x86)\Microsoft VS Code\bin;C:\Users\########\AppData\Local\atom\bin;C:\n\Users\########\AppData\Roaming\npm;C:\windows\system32
    10 verbose lifecycle ########@0.0.1~start: CWD: D:\########
    11 silly lifecycle ########@0.0.1~start: Args: [ '/d /s /c', 'electron .' ]
    12 silly lifecycle ########@0.0.1~start: Returned: code: 1  signal: null
    13 info lifecycle ########@0.0.1~start: Failed to exec start script
    14 verbose stack Error: ########@0.0.1 start: `electron .`
    14 verbose stack Exit status 1
    14 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:255:16)
    14 verbose stack     at emitTwo (events.js:106:13)
    14 verbose stack     at EventEmitter.emit (events.js:191:7)
    14 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:40:14)
    14 verbose stack     at emitTwo (events.js:106:13)
    14 verbose stack     at ChildProcess.emit (events.js:191:7)
    14 verbose stack     at maybeClose (internal/child_process.js:886:16)
    14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
    15 verbose pkgid ########@0.0.1
    16 verbose cwd D:\########
    17 error Windows_NT 10.0.14393
    18 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
    19 error node v6.10.3
    20 error npm  v3.10.10
    21 error code ELIFECYCLE
    22 error ########@0.0.1 start: `electron .`
    22 error Exit status 1
    23 error Failed at the ########@0.0.1 start script 'electron .'.
    23 error Make sure you have the latest version of node.js and npm installed.
    23 error If you do, this is most likely a problem with the ######## package,
    23 error not with npm itself.
    23 error Tell the author that this fails on your system:
    23 error     electron .
    23 error You can get information on how to open an issue for this project with:
    23 error     npm bugs ########
    23 error Or if that isn't available, you can get their info via:
    23 error     npm owner ls ########
    23 error There is likely additional logging output above.
    24 verbose exit [ 1, true ]

**Comment by MggMuggins** [comment]
More info can be found at this issue: https://github.com/electron/electron-quick-start/issues/30
I can also confirm this issue.

**Comment by gsilvestri** [comment]
I have exactly the same issue: I'm using VS Code to develop Angular JS application and I have exactly the same problem as VS Code is built on top of Atom.

The problem is that you have the same problem using the debugger tool in VS Code... so: **no Atom terminal means no debugger.**

Using the Ubuntu terminal instead the Atom/VS Code terminal a workaround, not a solution: needs to makes electron working with Atom terminal.

**Comment by SedesGobhani** [comment]
I also have this issue.
Win 8.1 x64
Atom v1.23.1 x64
Electron v1.8.2
platformio-ide-terminal v2.8.0 (Installed today.  Uninstalled today).
Tried some similar packages that fail in same way.
I'll just use independent command window.

**Comment by markobalogh2** [comment]
Are there any solutions to this other than using a command window? I need to launch my electron application from a node child_process `spawn` or `exec` call.

**Comment by Onfire7** [comment]
electron/electron#8200 Seems to be related.

Adding `set ELECTRON_RUN_AS_NODE=` as an auto run command should work around this issue, though platformio-atom-ide-terminal should probably strip that environment variable when appropriate, if possible... More info on exactly what's going on would greatly help come up with a proper solution...

As an aside, launching Electron projects from Hyper (https://github.com/zeit/hyper) seems to work just fine. I'm not really sure what they are doing to make that work unless they are always stripping out ELECTRON_RUN_AS_NODE...

**Comment by Minyar2004** [comment]
it works with platformio-ide-terminal 2.9.0 version

**Comment by the-j0k3r** [comment]
Yes, Im slowly going through similar reports, all have been fixed by 2.9.0
